feature-utils/poly-look/src/react-components/inputs/dropdown.jsx

```diff
@@ -0,0 +1,120 @@
+import { ChevronDown } from "../icons";
+
+import "./common.css";
+import "./dropdown.css";
+
+function getBodyClass(disabled, error, focused) {
+  if (disabled) return "body-disabled";
+  if ((error && focused) || focused) return "body-focused";
+  else if (error) return "body-error";
+}
+
+function filterOptions(options, text) {
+  return options.filter((option) => option.value.includes(text));
+}
+
+/** Dropdown with options filtering.
+ * @param {Object} props
+ * @param {Object} [props.initialSelection] - input value, has to contain
+ * an id and value, leave empty if the dropdown is not pre-populated
+ * @param {callback} [props.onChange] - onChange event handler
+ * @param {string} [props.name] - input name, also acts as id
+ * @param {string} [props.helperText] - input helper text
+ * @param {string} [props.label] - input label
+ * @param {number} [props.tabIndex] - input tabIndex, needed to make sure the
+ * focus/blur internal logic works
+ * @param {boolean} [props.error] - indicates if the input is in an error
+ * state or not
+ * @param {boolean} [props.disabled] - indicates if the input is disabled
+ * @param {Array[Object]} [props.options] - array containing the dropdown options,
+ * an option should have an id and a value
+ * @param {string} [props.noMatch] - text to be shown when there is no match
+ * @returns {JSX.Element}
+ */
+
+export function Dropdown({
+  initialSelection = {},
+  onChange,
+  name,
+  helperText,
+  label,
+  tabIndex,
+  error,
+  disabled,
+  options,
+  noMatch,
+}) {
+  const [open, setOpen] = useState(false);
+  const [dropdownOptions, setOptions] = useState(options);
+  const [query, setQuery] = useState(initialSelection.value || "");
+  const [selected, setSelected] = useState(initialSelection);
+  const inputRef = useRef();
+
+  let labelClass = query || open ? "label-up" : "";
+  let bodyClass = getBodyClass(disabled, error, open);
+
+  useEffect(() => {
+    onChange({ ...selected, name });
+  }, [selected]);
+
+  return (
+    <div className="poly-input-field poly-dropdown">
+      <div
+        data-testid="input-focusable"
+        className={`body ${bodyClass}`}
+        onFocus={() => {
+          setOpen(true);
+          inputRef.current.focus();
+        }}
+        onBlur={() => setOpen(false)}
+        tabIndex={tabIndex}
+      >
+        <label htmlFor={name} className={labelClass}>
+          {label}
+        </label>
+        <input
+          disabled={disabled}
+          autoComplete="off"
+          ref={inputRef}
+          onChange={(e) => {
+            setOptions(filterOptions(options, e.target.value));
+            setQuery(e.target.value);
+            setSelected({ id: null, value: null });
+          }}
+          name={name}
+          id={name}
+          value={query}
+        />
+        <ChevronDown
+          fill="var(--pl-field-trailing-icon-color)"
+          className={`poly-icon-small icon-${open ? "up" : "down"}`}
```

Comment:
Icon should be 24px not 16px.

Would it be better to add a general class to the icons with a size token instead of adding the `poly-icon-small` class to each? 
Like that, if the size changes, it will be a general change in all field icons, that way you don't need to update the class of the component but just a token value.

Something like
`
pl-field-icon-size: 24px;

width: pl-field-icon-size;
height: pl-field-icon-size;
`

Reply:
I've added the token and switched to a generic icon class for the components.